RFS/src/main/java/org/opensearch/migrations/bulkload/common/InvalidResponse.java

```diff
@@ -17,7 +17,7 @@
 
 @Slf4j
 public class InvalidResponse extends RfsException {
-    private static final Pattern UNKNOWN_SETTING = Pattern.compile("unknown setting \\[(.+?)\\].+");
+    private static final Pattern UNKNOWN_SETTING = Pattern.compile("unknown setting \\[([a-zA-Z0-9_.-]+)\\].+");
```

Comment:
Would the issue also be resolved by changing `.+?` -> `.+`

Reply:
Yes, but that expression wouldn't be correct since it could match the closing bracket or a host of invalid characters that opensearch would never use in a setting name.
- ✔️ `unknown setting [foo.bar] other stuff` -> `foo.bar` 
- ❌ `unknown setting [foo.bar] other stuff ]` -> `foo.bar] other stuff`

Note, this could also be mitigated by removing the trailing `.+` match, but this version is more correct and CI is nearly passed.

Reply:
I see.. the issue is that `]` shouldn't be a matching character in the inner match.

`"unknown setting \\[([^\]\n]+?)\\].+"` this would be the most similar to the original logic that doesn't have the bug

This change seems safe to set the characters inside the `[]` to those in this PR